src/state/slices/postsSlice.ts

```diff
@@ -14,31 +16,98 @@ export type Post = {
   parentPostId?: number;
 };
 
-const initialState: Post[] = [];
+type PostSliceType = {
+  posts: Post[];
+  expandedPost: Post;
+};
+
+const initialState: PostSliceType = {
+  posts: [],
+  expandedPost: {
+    displayName: "",
+    isLikedByCurrentUser: false,
+    numberOfLikes: 0,
+    numberOfReplies: 0,
+    numberOfReposts: 0,
+    postId: 0,
+    textContent: "",
+    timestamp: "",
+    username: "",
+    isRepost: false,
+    isQuotePost: false,
+  },
+};
 
 export const postsSlice = createSlice({
   name: "post",
   initialState,
   reducers: {
-    setPosts: (_, action: PayloadAction<Post[]>) => {
-      return action.payload;
-    },
-    updatePost: (state, action: PayloadAction<Post>) => {
-      const newPosts = state.map((o) => {
-        if (o.postId === action.payload.postId) {
-          return action.payload;
+    addReply: (state, action: PayloadAction<Post>) => {
+      const newReply = {
```

Comment:
Nice work, since you've included `numberofReposts` maybe we should add this to the `addReply` and `appendPost` reducers similar to what you've done with `numberofReplies`.

Though it is considered out of scope and could be done in another issue that has to do with the reposting functionality
```suggestion
      const newReply = {
          ...action.payload,
          numberOfLikes: 0,
          numberOfReplies: 0,
          numberOfReposts: 0,
      };
```

Reply:
Yea I forgot about this, I'll add it in